discoverd/deployment/deployment.go

```diff
@@ -88,6 +89,21 @@ outer:
 		if err := d.update(); err != nil {
 			return err
 		}
+		// If the metadata is nil then wait for another metadata event before re-evaluating
+		if d.meta == nil {
+			select {
+			case event, ok := <-d.events:
+				if !ok {
+					return fmt.Errorf("service stream closed unexpectedly: %s", d.stream.Err())
+				}
+				if event.Kind == discoverd.EventKindServiceMeta {
+					d.meta = event.ServiceMeta
+					continue outer
```

Comment:
It seems a bit weird to call `update` again, which this appears to do. Shouldn't this break out if meta is non-nil?